/**
 * This file is part of the "libfnord" project
 *   Copyright (c) 2015 Paul Asmuth
 *
 * FnordMetric is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License v3.0. You should have received a
 * copy of the GNU General Public License along with this program. If not, see
 * <http://www.gnu.org/licenses/>.
 */
package tsdb;

message TableConfig {

  /**
   * The tsdb namespace to be configured
   */
  required string tsdb_namespace = 6;

  /**
   * The full path/name of the schema
   */
  required string schema = 7;

  /**
   * The stream key prefix for which this config should be applied/. (empty
   * matches all streams, longest prefix wins)
   */
  optional string stream_key_prefix = 1;

  /**
   * The maximum sstable size in bytes
   */
  required uint64 sstable_size = 4;

  /**
   * The compaction interval in microseconds
   */
  optional uint64 compaction_interval = 5;

};
